src/Persistence_matrix/include/gudhi/persistence_matrix_options.h

```diff
@@ -0,0 +1,169 @@
+/*    This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+ *    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+ *    Author(s):       Hannah Schreiber
+ *
+ *    Copyright (C) 2022-24 Inria
+ *
+ *    Modification(s):
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+/**
+ * @file persistence_matrix_options.h
+ * @author Hannah Schreiber
+ * @brief Contains the options for the matrix template.
+ */
+
+#ifndef PM_OPTIONS_INCLUDED
+#define PM_OPTIONS_INCLUDED
+
+#include "Fields/Zp_field_operators.h"
+
+namespace Gudhi {
+namespace persistence_matrix {
+
+/**
+ * @ingroup persistence_matrix
+ *
+ * @brief List of column types.
+ */
+enum Column_types { 
```

Comment:
Is it possible for a user to use their own column type? There is a concept of what such a column type must look like, but from the use of this enum, it doesn't look like a user has any way of telling matrix to use it.